recorder.go

```diff
@@ -377,29 +378,18 @@ func convertToKeyValue(k string, value interface{}) *cpb.KeyValue {
 	return &kv
 }
 
-// TODO: Decide what we want to do about duplicate keys
-func translatePayload(pl interface{}, kvs []*cpb.KeyValue) []*cpb.KeyValue {
-	switch p := pl.(type) {
-	case map[string]interface{}:
-		for k, v := range p {
-			kvs = append(kvs, convertToKeyValue(k, v))
-		}
-	case []interface{}:
-		for _, v := range p {
-			kvs = append(kvs, convertToKeyValue(payloadKey, v))
-		}
-	case nil:
-		return nil
-	default:
-		kvs = append(kvs, convertToKeyValue(payloadKey, p))
-	}
-	return kvs
-}
-
 func translateEventAndPayload(e string, pl interface{}) []*cpb.KeyValue {
 	var kvs []*cpb.KeyValue
 	kvs = append(kvs, &cpb.KeyValue{Key: messageKey, Value: &cpb.KeyValue_StringValue{e}})
```

Comment:
Nit: I'd probably write this without an append, e.g., a one-liner:

kvs := []*cpb.KeyValue{&cpb.KeyValue{Key: ..., Value: ...}}